### PR TITLE
nglib 5.3.1 (new formula)

### PR DESCRIPTION
**Pull request by ianrrees**
nglib is a tetrahedral meshing library, which we use in FreeCAD to generate meshes for finite element simulations.

### Have you:

- [x] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [x] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [x] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [x] Built your formula locally prior to submission with `brew install <formula>`?

### New formula: have you

- [x] Written a sensible test? (The best test is to compile and run a sample program.)

### Updates to existing formula: have you

- [ ] Removed the `revision` line, if any, when bumping the version number?
- [ ] Added/bumped the `revision` number if the changes affect the precompiled bottles?
- [ ] Not altered the `bottle` section?
- [ ] Checked that the tests still pass?

### CODE REVIEW

**Comment by ianrrees** [comment]
Changed the download URL to pass the audit AND resolve correctly!

FWIW - Hadn't caught this before submitting as the original download URL I used for development worked, but then the audit tool suggested changing http to https and using downloads.sourceforge.  Unfortunately, the rest of the URL is subtly different in that case, but homebrew's cache already had the right file with the same name...

**Comment by ianrrees** [comment]
I suspect the CI failure is due to one of the patches having windows-style line endings, where the others have unix ones.  What's the preferred way to fix this?

**Comment by ianrrees** [comment]
sed to the rescue!  I couldn't figure out a way to say "if Formula["opencascade"].version >= 6.8.1" to put around the sed replacements, but can't imagine there's much interest in nglib with old versions of opencascade.

File: nglib.rb [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      }
+      int main(int argc, char **argv) {
+          nglib::Ng_Init();
+          auto mesh(nglib::Ng_NewMesh());
+          nglib::Ng_DeleteMesh(mesh);
+          nglib::Ng_Exit();
+          return 0;
+      }
+    EOS
+    system ENV.cxx, "-Wall", "-o", "test", "test.cpp", "-std=c++11",
```
Comment (dpo) [inline nglib.rb]: Does the library need to be compiled in c++11 mode?

Comment (ianrrees) [inline nglib.rb]: No, the library is all C++98/03 AFAICT.

I only used C++11 for the test out of habit, could change the "auto" to "nglib::Ng_Mesh" if it's better to compile the test case without C++11 support?

Comment (dpo) [inline nglib.rb]: It might be simpler to remove c++11 support. For a moment I wondered if the entire formula needed to depend on `:cxx11`.

Comment (ianrrees) [inline nglib.rb]: @dpo - sorry if you've already gotten a notification for this, just wanted to mention that I changed the test to use default C++ standard.  Thanks!

**Comment by dpo** [comment]
Thanks!

**Pull request closed without merging**